ui/page/components/utils.go

```diff
@@ -12,3 +17,24 @@ func ContextDone(ctx context.Context) bool {
 		return false
 	}
 }
+
+// RetryFunc implement retry policy for processes that needs to be executed
+// the after initial failure.
+func RetryFunc(retryAttempts int, sleepDur time.Duration, funcDesc string, errFunc func() error) (int, error) {
+	var err error
+	for i := 0; i < retryAttempts; i++ {
+		if i > 0 {
+			if i > 1 {
+				sleepDur *= 2
+			}
+			log.Printf("waiting %s to retry function %s after error: %v", sleepDur, funcDesc, err)
```

Comment:
Just seeing this, `log` is an imported package. Not sure how it works but it certainly does not use the same logging pattern I've seen in the rest of this godcr codebase. I don't think the components pkg has a logger yet, I think #777 introduces one. This can be `fmt.Printf` for now and should end with a `\n`.